src/uu/split/src/split.rs

```diff
@@ -52,14 +54,127 @@ const AFTER_HELP: &str = help_section!("after help", "split.md");
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let (args, obs_lines) = handle_obsolete(args);
+
     let matches = uu_app().try_get_matches_from(args)?;
-    match Settings::from(&matches) {
+
+    match Settings::from(&matches, &obs_lines) {
         Ok(settings) => split(&settings),
         Err(e) if e.requires_usage() => Err(UUsageError::new(1, format!("{e}"))),
         Err(e) => Err(USimpleError::new(1, format!("{e}"))),
     }
 }
 
+/// Extract obsolete shorthand (if any) for specifying lines in following scenarios (and similar)
+/// `split -22 file` would mean `split -l 22 file`
+/// `split -2de file` would mean `split -l 2 -d -e file`
+/// `split -x300e file` would mean `split -x -l 300 -e file`
+/// `split -x300e -22 file` would mean `split -x -e -l 22 file` (last obsolete lines option wins)
+/// following GNU `split` behavior
+fn handle_obsolete(args: impl uucore::Args) -> (Vec<OsString>, Option<String>) {
```

Comment:
This function is a bit too long
What about something like:
```
fn handle_obsolete(args: impl uucore::Args) -> (Vec<OsString>, Option<String>) {
    let mut obs_lines = None;
    let mut preceding_long_opt_req_value = false;
    let mut preceding_short_opt_req_value = false;

    let filtered_args = args
        .filter_map(|os_slice| filter_args(
            os_slice,
            &mut obs_lines,
            &mut preceding_long_opt_req_value,
            &mut preceding_short_opt_req_value,
        ))
        .collect();

    (filtered_args, obs_lines)
}

fn filter_args(
    os_slice: OsString,
    obs_lines: &mut Option<String>,
    preceding_long_opt_req_value: &mut bool,
    preceding_short_opt_req_value: &mut bool,
) -> Option<OsString> {
    let filter: Option<OsString>;
    if let Some(slice) = os_slice.to_str() {
        if should_extract_obs_lines(slice, preceding_long_opt_req_value, preceding_short_opt_req_value) {
            filter = handle_extract_obs_lines(slice, obs_lines);
        } else {
            filter = Some(OsString::from(slice));
        }
        handle_preceding_options(slice, preceding_long_opt_req_value, preceding_short_opt_req_value);
    } else {
        filter = Some(os_slice);
    }
    filter
}

fn should_extract_obs_lines(slice: &str, preceding_long_opt_req_value: &bool, preceding_short_opt_req_value: &bool) -> bool {
    slice.starts_with('-')
        && !slice.starts_with("--")
        && !preceding_long_opt_req_value
        && !preceding_short_opt_req_value
        && !slice.starts_with("-a")
        && !slice.starts_with("-b")
        && !slice.starts_with("-C")
        && !slice.starts_with("-l")
        && !slice.starts_with("-n")
}

fn handle_extract_obs_lines(slice: &str, obs_lines: &mut Option<String>) -> Option<OsString> {
    // ... (existing logic for extracting obsolete lines)
}

fn handle_preceding_options(slice: &str, preceding_long_opt_req_value: &mut bool, preceding_short_opt_req_value: &mut bool) {
    // ... (existing logic for handling preceding options)
}
```

Reply:
yep, will refactor